src/inference/src/dev/iplugin.cpp

```diff
@@ -169,36 +225,193 @@ std::unordered_set<std::string> ov::get_supported_nodes(
+                                    split_node_set.insert(op->get_friendly_name());
+                                    // Judge if the current constant op should be removed from supported
+                                    if (total_size < max_query_size)
+                                        continue;
+                                }
+                            }
+                        }
+                    }
+                    // Start splitting when ready and the ops is constant
+                    if (ready_split) {
+                        if (ov::op::util::is_constant(op)) {
+                            remove_op_from_supported(op);
+                            start_split = true;
+                        } else if (start_split) {
+                            remove_op_from_supported(op);
+                            for (auto& input : op->inputs()) {
+                                const auto& node = get_input_node(input);
+                                if (ov::op::util::is_constant(node)) {
+                                    remove_op_from_supported(node);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            // In case some ops not in orederd, so traverse the entire model to ensure accurate split
+            bool changed = true;
+            while (changed) {
+                changed = false;
+                for (auto& op : model->get_ordered_ops()) {
+                    if (!supported.count(op->get_friendly_name()) && has_users_supported(supported, op) &&
+                        !unsupported.count(op->get_friendly_name())) {
+                        supported.insert(op->get_friendly_name());
+                        changed = true;
+                    }
+                }
+            }
+            // Calculate the data size that needs to be transmitted after the current model is split
+            int64_t total_len = 0;
+            for (auto& op : model->get_ordered_ops()) {
+                if (supported.count(op->get_friendly_name()) && !ov::op::util::is_constant(op) &&
+                    !ov::op::util::is_parameter(op)) {
+                    if (has_users_unsupported(supported, op)) {
+                        int64_t op_size = 1;
+                        for (size_t shape_id = 0; shape_id < op->get_output_partial_shape(0).size(); shape_id++) {
+                            if (!op->get_output_partial_shape(0)[shape_id].is_dynamic()) {
+                                int64_t len = op->get_output_partial_shape(0)[shape_id].get_length();
+                                if (len >= 1)
+                                    op_size *= len;
+                            }
+                        }
+                        total_len += op_size;
+                    }
+                }
+            }
+            if ((total_len < last_total_len || last_total_len == 0) && !cancel_split) {
```

Comment:
can you paste perf data here on the benefit of this least data transmission searching effort?

Reply:
![image](https://github.com/openvinotoolkit/openvino/assets/31067432/521c4bad-6f4a-4ff3-8037-362e96a24580)